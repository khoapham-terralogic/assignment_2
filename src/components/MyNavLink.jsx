import React from 'react'
import PropTypes from 'prop-types'
import { NavLink } from 'react-router-dom';

const MyNavLink = ({ lable, toPath, onClick }) => {
    return (
        <NavLink
            to={toPath}
            onClick={onClick}
            className="nav-link nav-link-custom"
        >
            <div className="nav-item">{lable}</div>
        </NavLink>
    );
}

MyNavLink.propTypes = {
    lable: PropTypes.string.isRequired,
    className: PropTypes.string,
    toPath: PropTypes.string.isRequired,
    onClick: PropTypes.func
}

MyNavLink.defaultProps = {
    lable: "",
    className: "",
    toPath: "/"
}

export default MyNavLink;
