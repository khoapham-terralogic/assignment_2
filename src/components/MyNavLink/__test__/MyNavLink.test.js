import React from 'react';
import { shallow } from 'enzyme';
import MyNavLink from '..'

const props = {
    lable: "Register",
    toPath: "/auth/register",
    onClick: jest.fn()
}

describe("Test <MyNavLink1 />", () => {
    const wrapper = shallow(<MyNavLink {...props} />)
    it("It should render", () => {
        expect(wrapper).toMatchSnapshot()
    })
    it("It should contain props", () => {
        expect(wrapper.find(".nav-item").text()).toBe("Register");
        expect(wrapper.prop('to')).toEqual("/auth/register");
    })
})