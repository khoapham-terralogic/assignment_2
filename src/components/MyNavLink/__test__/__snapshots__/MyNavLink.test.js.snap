// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <MyNavLink1 /> It should render 1`] = `
<NavLink
  className="nav-link nav-link-custom"
  data-testid="navlink"
  onClick={[MockFunction]}
  to="/auth/register"
>
  <div
    className="nav-item"
  >
    Register
  </div>
</NavLink>
`;
